org.scala-ide.sdt.core/src/org/scalaide/core/IScalaProject.scala

```diff
@@ -222,6 +224,13 @@ trait IScalaProject extends Publisher[IScalaProjectEvent] {
    *  @param workingCopyOwner A working copy owner who's copies are searched first.
    */
   def newSearchableEnvironment(workingCopyOwner: WorkingCopyOwner = DefaultWorkingCopyOwner.PRIMARY): SearchableEnvironment
+
+  /**
+   * Says if compilation should be done in scopes e.g. main classes should be compiled
+   * before test classes. Used when compilation error in some scope should not break
+   * the compilation of dependent projects.
+   */
+  def compileInScopes(decision: Boolean): Unit
```

Comment:
Well, generally it can be eventually removed. It has been introduced for project with let me call it 'not nice source-output structure' where output it shared by different scopes. Let me think about it a little bit... Anyway I wish the sbt compiler would resolve all dependencies itself and just give back the compilation result with cool API just to create markers